system/jlib/jqueue.hpp

```diff
@@ -546,5 +546,54 @@ class DListOf
     }
 };
 
+template <typename T>
+class CSPSCQueue
```

Comment:
This name should could be clearer - or at least preceded with a comment Single Producer, Single Consumer I assume